AppServer/google/appengine/api/logservice/logservice_stub.py

```diff
@@ -238,6 +348,59 @@ def end_request(self, request_id, status, response_size, end_time=None):
     rl.responseSize = response_size
     rl.endTime = end_time
     self._pending_requests_applogs[request_id].finish()
+
+    if self.is_elk_enabled:
+      start_time = rl.startTime
+      start_time_ms = float(start_time) / 1000
+      end_time_ms = float(end_time) / 1000
+
+      # Render app logs:
+      try:
+        app_logs_str = u'\n'.join([
+          u'{} {} {}'.format(
+            LEVELS[log.level],
+            datetime.utcfromtimestamp(log.time/1000000)
+              .strftime('%Y-%m-%d %H:%M:%S'),
+            log.message
+          )
+          for log in rl.appLogs
+        ])
+      except UnicodeError:
+        app_logs_str = u'\n'.join([
+          u'{} {} {}'.format(
+            LEVELS[log.level],
+            datetime.utcfromtimestamp(log.time/1000000)
+              .strftime('%Y-%m-%d %H:%M:%S'),
+            unicode(log.message, 'ascii', 'ignore')
```

Comment:
It seems like you could place this above (or just the byte string's .decode method) and omit the try since you are coercing bytes to unicode above anyway. I would also recommend using the 'utf-8' codec to avoid stripping non-ascii characters.